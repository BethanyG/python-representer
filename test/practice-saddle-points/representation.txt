Module([FunctionDef('placeholder_0', arguments([], [arg('placeholder_1', lineno=1, col_offset=18, end_lineno=1, end_col_offset=24)], kwonlyargs=[], kw_defaults=[], defaults=[]), [If(UnaryOp(Not(), Name('placeholder_1', Load(), lineno=2, col_offset=11, end_lineno=2, end_col_offset=17), lineno=2, col_offset=7, end_lineno=2, end_col_offset=17), [Return(List([], Load(), lineno=3, col_offset=15, end_lineno=3, end_col_offset=17), lineno=3, col_offset=8, end_lineno=3, end_col_offset=17)], [], lineno=2, col_offset=4, end_lineno=3, end_col_offset=17), If(Call(Name('any', Load(), lineno=5, col_offset=7, end_lineno=5, end_col_offset=10), [GeneratorExp(Compare(Call(Name('len', Load(), lineno=5, col_offset=11, end_lineno=5, end_col_offset=14), [Name('placeholder_2', Load(), lineno=5, col_offset=15, end_lineno=5, end_col_offset=18)], [], lineno=5, col_offset=11, end_lineno=5, end_col_offset=19), [NotEq()], [Call(Name('len', Load(), lineno=5, col_offset=23, end_lineno=5, end_col_offset=26), [Subscript(Name('placeholder_1', Load(), lineno=5, col_offset=27, end_lineno=5, end_col_offset=33), Constant(0, lineno=5, col_offset=34, end_lineno=5, end_col_offset=35), Load(), lineno=5, col_offset=27, end_lineno=5, end_col_offset=36)], [], lineno=5, col_offset=23, end_lineno=5, end_col_offset=37)], lineno=5, col_offset=11, end_lineno=5, end_col_offset=37), [comprehension(Name('placeholder_2', Store(), lineno=5, col_offset=42, end_lineno=5, end_col_offset=45), Name('placeholder_1', Load(), lineno=5, col_offset=49, end_lineno=5, end_col_offset=55), [], 0)], lineno=5, col_offset=10, end_lineno=5, end_col_offset=56)], [], lineno=5, col_offset=7, end_lineno=5, end_col_offset=56), [Raise(Call(Name('ValueError', Load(), lineno=6, col_offset=14, end_lineno=6, end_col_offset=24), [Constant('irregular matrix', lineno=6, col_offset=25, end_lineno=6, end_col_offset=43)], [], lineno=6, col_offset=14, end_lineno=6, end_col_offset=44), lineno=6, col_offset=8, end_lineno=6, end_col_offset=44)], [], lineno=5, col_offset=4, end_lineno=6, end_col_offset=44), Assign([Name('placeholder_3', Store(), lineno=8, col_offset=4, end_lineno=8, end_col_offset=8)], ListComp(Call(Name('max', Load(), lineno=8, col_offset=12, end_lineno=8, end_col_offset=15), [Name('placeholder_2', Load(), lineno=8, col_offset=16, end_lineno=8, end_col_offset=19)], [], lineno=8, col_offset=12, end_lineno=8, end_col_offset=20), [comprehension(Name('placeholder_2', Store(), lineno=8, col_offset=25, end_lineno=8, end_col_offset=28), Name('placeholder_1', Load(), lineno=8, col_offset=32, end_lineno=8, end_col_offset=38), [], 0)], lineno=8, col_offset=11, end_lineno=8, end_col_offset=39), lineno=8, col_offset=4, end_lineno=8, end_col_offset=39), Assign([Name('placeholder_4', Store(), lineno=9, col_offset=4, end_lineno=9, end_col_offset=8)], ListComp(Call(Name('min', Load(), lineno=9, col_offset=12, end_lineno=9, end_col_offset=15), [Name('placeholder_5', Load(), lineno=9, col_offset=16, end_lineno=9, end_col_offset=19)], [], lineno=9, col_offset=12, end_lineno=9, end_col_offset=20), [comprehension(Name('placeholder_5', Store(), lineno=9, col_offset=25, end_lineno=9, end_col_offset=28), Call(Name('zip', Load(), lineno=9, col_offset=32, end_lineno=9, end_col_offset=35), [Starred(Name('placeholder_1', Load(), lineno=9, col_offset=37, end_lineno=9, end_col_offset=43), Load(), lineno=9, col_offset=36, end_lineno=9, end_col_offset=43)], [], lineno=9, col_offset=32, end_lineno=9, end_col_offset=44), [], 0)], lineno=9, col_offset=11, end_lineno=9, end_col_offset=45), lineno=9, col_offset=4, end_lineno=9, end_col_offset=45), Assign([Name('placeholder_6', Store(), lineno=11, col_offset=4, end_lineno=11, end_col_offset=10)], ListComp(Dict([Constant('row', lineno=11, col_offset=15, end_lineno=11, end_col_offset=20), Constant('column', lineno=11, col_offset=33, end_lineno=11, end_col_offset=41)], [BinOp(Name('placeholder_7', Load(), lineno=11, col_offset=22, end_lineno=11, end_col_offset=27), Add(), Constant(1, lineno=11, col_offset=30, end_lineno=11, end_col_offset=31), lineno=11, col_offset=22, end_lineno=11, end_col_offset=31), BinOp(Name('placeholder_9', Load(), lineno=11, col_offset=43, end_lineno=11, end_col_offset=52), Add(), Constant(1, lineno=11, col_offset=55, end_lineno=11, end_col_offset=56), lineno=11, col_offset=43, end_lineno=11, end_col_offset=56)], lineno=11, col_offset=14, end_lineno=11, end_col_offset=57), [comprehension(Tuple([Name('placeholder_7', Store(), lineno=12, col_offset=18, end_lineno=12, end_col_offset=23), Name('placeholder_8', Store(), lineno=12, col_offset=25, end_lineno=12, end_col_offset=26)], Store(), lineno=12, col_offset=18, end_lineno=12, end_col_offset=26), Call(Name('enumerate', Load(), lineno=12, col_offset=30, end_lineno=12, end_col_offset=39), [Name('placeholder_1', Load(), lineno=12, col_offset=40, end_lineno=12, end_col_offset=46)], [], lineno=12, col_offset=30, end_lineno=12, end_col_offset=47), [], 0), comprehension(Tuple([Name('placeholder_9', Store(), lineno=13, col_offset=18, end_lineno=13, end_col_offset=27), Name('placeholder_8', Store(), lineno=13, col_offset=29, end_lineno=13, end_col_offset=30)], Store(), lineno=13, col_offset=18, end_lineno=13, end_col_offset=30), Call(Name('enumerate', Load(), lineno=13, col_offset=34, end_lineno=13, end_col_offset=43), [Subscript(Name('placeholder_1', Load(), lineno=13, col_offset=44, end_lineno=13, end_col_offset=50), Constant(0, lineno=13, col_offset=51, end_lineno=13, end_col_offset=52), Load(), lineno=13, col_offset=44, end_lineno=13, end_col_offset=53)], [], lineno=13, col_offset=34, end_lineno=13, end_col_offset=54), [Compare(Subscript(Name('placeholder_3', Load(), lineno=14, col_offset=17, end_lineno=14, end_col_offset=21), Name('placeholder_7', Load(), lineno=14, col_offset=22, end_lineno=14, end_col_offset=27), Load(), lineno=14, col_offset=17, end_lineno=14, end_col_offset=28), [Eq()], [Subscript(Name('placeholder_4', Load(), lineno=14, col_offset=32, end_lineno=14, end_col_offset=36), Name('placeholder_9', Load(), lineno=14, col_offset=37, end_lineno=14, end_col_offset=46), Load(), lineno=14, col_offset=32, end_lineno=14, end_col_offset=47)], lineno=14, col_offset=17, end_lineno=14, end_col_offset=47)], 0)], lineno=11, col_offset=13, end_lineno=14, end_col_offset=48), lineno=11, col_offset=4, end_lineno=14, end_col_offset=48), Return(BoolOp(Or(), [Name('placeholder_6', Load(), lineno=16, col_offset=11, end_lineno=16, end_col_offset=17), List([], Load(), lineno=16, col_offset=21, end_lineno=16, end_col_offset=23)], lineno=16, col_offset=11, end_lineno=16, end_col_offset=23), lineno=16, col_offset=4, end_lineno=16, end_col_offset=23)], [], lineno=1, col_offset=0, end_lineno=16, end_col_offset=23)], [])