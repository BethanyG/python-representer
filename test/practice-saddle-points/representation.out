## BEGIN NORMALIZED CODE ##
def placeholder_0(placeholder_1):
    if not placeholder_1:
        return []
    if any(
        (len(placeholder_2) != len(placeholder_1[0]) for placeholder_2 in placeholder_1)
    ):
        raise ValueError("irregular matrix")
    placeholder_3 = [max(placeholder_2) for placeholder_2 in placeholder_1]
    placeholder_4 = [min(placeholder_5) for placeholder_5 in zip(*placeholder_1)]
    placeholder_6 = [
        {"row": placeholder_7 + 1, "column": placeholder_9 + 1}
        for placeholder_7, placeholder_8 in enumerate(placeholder_1)
        for placeholder_9, placeholder_8 in enumerate(placeholder_1[0])
        if placeholder_3[placeholder_7] == placeholder_4[placeholder_9]
    ]
    return placeholder_6 or []

## END NORMALIZED CODE ##

## BEGIN TREE BEFORE ##
Module(
  [
    FunctionDef(
      'saddle_points',
      arguments(
        [],
        [
          arg(
            'matrix',
            lineno=1,
            col_offset=18,
            end_lineno=1,
            end_col_offset=24)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        If(
          UnaryOp(
            Not(),
            Name(
              'matrix',
              Load(),
              lineno=2,
              col_offset=11,
              end_lineno=2,
              end_col_offset=17),
            lineno=2,
            col_offset=7,
            end_lineno=2,
            end_col_offset=17),
          [
            Return(
              List(
                [],
                Load(),
                lineno=3,
                col_offset=15,
                end_lineno=3,
                end_col_offset=17),
              lineno=3,
              col_offset=8,
              end_lineno=3,
              end_col_offset=17)],
          [],
          lineno=2,
          col_offset=4,
          end_lineno=3,
          end_col_offset=17),
        If(
          Call(
            Name(
              'any',
              Load(),
              lineno=5,
              col_offset=7,
              end_lineno=5,
              end_col_offset=10),
            [
              GeneratorExp(
                Compare(
                  Call(
                    Name(
                      'len',
                      Load(),
                      lineno=5,
                      col_offset=11,
                      end_lineno=5,
                      end_col_offset=14),
                    [
                      Name(
                        'row',
                        Load(),
                        lineno=5,
                        col_offset=15,
                        end_lineno=5,
                        end_col_offset=18)],
                    [],
                    lineno=5,
                    col_offset=11,
                    end_lineno=5,
                    end_col_offset=19),
                  [
                    NotEq()],
                  [
                    Call(
                      Name(
                        'len',
                        Load(),
                        lineno=5,
                        col_offset=23,
                        end_lineno=5,
                        end_col_offset=26),
                      [
                        Subscript(
                          Name(
                            'matrix',
                            Load(),
                            lineno=5,
                            col_offset=27,
                            end_lineno=5,
                            end_col_offset=33),
                          Constant(
                            0,
                            lineno=5,
                            col_offset=34,
                            end_lineno=5,
                            end_col_offset=35),
                          Load(),
                          lineno=5,
                          col_offset=27,
                          end_lineno=5,
                          end_col_offset=36)],
                      [],
                      lineno=5,
                      col_offset=23,
                      end_lineno=5,
                      end_col_offset=37)],
                  lineno=5,
                  col_offset=11,
                  end_lineno=5,
                  end_col_offset=37),
                [
                  comprehension(
                    Name(
                      'row',
                      Store(),
                      lineno=5,
                      col_offset=42,
                      end_lineno=5,
                      end_col_offset=45),
                    Name(
                      'matrix',
                      Load(),
                      lineno=5,
                      col_offset=49,
                      end_lineno=5,
                      end_col_offset=55),
                    [],
                    0)],
                lineno=5,
                col_offset=10,
                end_lineno=5,
                end_col_offset=56)],
            [],
            lineno=5,
            col_offset=7,
            end_lineno=5,
            end_col_offset=56),
          [
            Raise(
              Call(
                Name(
                  'ValueError',
                  Load(),
                  lineno=6,
                  col_offset=14,
                  end_lineno=6,
                  end_col_offset=24),
                [
                  Constant(
                    'irregular matrix',
                    lineno=6,
                    col_offset=25,
                    end_lineno=6,
                    end_col_offset=43)],
                [],
                lineno=6,
                col_offset=14,
                end_lineno=6,
                end_col_offset=44),
              lineno=6,
              col_offset=8,
              end_lineno=6,
              end_col_offset=44)],
          [],
          lineno=5,
          col_offset=4,
          end_lineno=6,
          end_col_offset=44),
        Assign(
          [
            Name(
              'mmax',
              Store(),
              lineno=8,
              col_offset=4,
              end_lineno=8,
              end_col_offset=8)],
          ListComp(
            Call(
              Name(
                'max',
                Load(),
                lineno=8,
                col_offset=12,
                end_lineno=8,
                end_col_offset=15),
              [
                Name(
                  'row',
                  Load(),
                  lineno=8,
                  col_offset=16,
                  end_lineno=8,
                  end_col_offset=19)],
              [],
              lineno=8,
              col_offset=12,
              end_lineno=8,
              end_col_offset=20),
            [
              comprehension(
                Name(
                  'row',
                  Store(),
                  lineno=8,
                  col_offset=25,
                  end_lineno=8,
                  end_col_offset=28),
                Name(
                  'matrix',
                  Load(),
                  lineno=8,
                  col_offset=32,
                  end_lineno=8,
                  end_col_offset=38),
                [],
                0)],
            lineno=8,
            col_offset=11,
            end_lineno=8,
            end_col_offset=39),
          lineno=8,
          col_offset=4,
          end_lineno=8,
          end_col_offset=39),
        Assign(
          [
            Name(
              'mmin',
              Store(),
              lineno=9,
              col_offset=4,
              end_lineno=9,
              end_col_offset=8)],
          ListComp(
            Call(
              Name(
                'min',
                Load(),
                lineno=9,
                col_offset=12,
                end_lineno=9,
                end_col_offset=15),
              [
                Name(
                  'col',
                  Load(),
                  lineno=9,
                  col_offset=16,
                  end_lineno=9,
                  end_col_offset=19)],
              [],
              lineno=9,
              col_offset=12,
              end_lineno=9,
              end_col_offset=20),
            [
              comprehension(
                Name(
                  'col',
                  Store(),
                  lineno=9,
                  col_offset=25,
                  end_lineno=9,
                  end_col_offset=28),
                Call(
                  Name(
                    'zip',
                    Load(),
                    lineno=9,
                    col_offset=32,
                    end_lineno=9,
                    end_col_offset=35),
                  [
                    Starred(
                      Name(
                        'matrix',
                        Load(),
                        lineno=9,
                        col_offset=37,
                        end_lineno=9,
                        end_col_offset=43),
                      Load(),
                      lineno=9,
                      col_offset=36,
                      end_lineno=9,
                      end_col_offset=43)],
                  [],
                  lineno=9,
                  col_offset=32,
                  end_lineno=9,
                  end_col_offset=44),
                [],
                0)],
            lineno=9,
            col_offset=11,
            end_lineno=9,
            end_col_offset=45),
          lineno=9,
          col_offset=4,
          end_lineno=9,
          end_col_offset=45),
        Assign(
          [
            Name(
              'points',
              Store(),
              lineno=11,
              col_offset=4,
              end_lineno=11,
              end_col_offset=10)],
          ListComp(
            Dict(
              [
                Constant(
                  'row',
                  lineno=11,
                  col_offset=15,
                  end_lineno=11,
                  end_col_offset=20),
                Constant(
                  'column',
                  lineno=11,
                  col_offset=33,
                  end_lineno=11,
                  end_col_offset=41)],
              [
                BinOp(
                  Name(
                    'index',
                    Load(),
                    lineno=11,
                    col_offset=22,
                    end_lineno=11,
                    end_col_offset=27),
                  Add(),
                  Constant(
                    1,
                    lineno=11,
                    col_offset=30,
                    end_lineno=11,
                    end_col_offset=31),
                  lineno=11,
                  col_offset=22,
                  end_lineno=11,
                  end_col_offset=31),
                BinOp(
                  Name(
                    'col_index',
                    Load(),
                    lineno=11,
                    col_offset=43,
                    end_lineno=11,
                    end_col_offset=52),
                  Add(),
                  Constant(
                    1,
                    lineno=11,
                    col_offset=55,
                    end_lineno=11,
                    end_col_offset=56),
                  lineno=11,
                  col_offset=43,
                  end_lineno=11,
                  end_col_offset=56)],
              lineno=11,
              col_offset=14,
              end_lineno=11,
              end_col_offset=57),
            [
              comprehension(
                Tuple(
                  [
                    Name(
                      'index',
                      Store(),
                      lineno=12,
                      col_offset=18,
                      end_lineno=12,
                      end_col_offset=23),
                    Name(
                      '_',
                      Store(),
                      lineno=12,
                      col_offset=25,
                      end_lineno=12,
                      end_col_offset=26)],
                  Store(),
                  lineno=12,
                  col_offset=18,
                  end_lineno=12,
                  end_col_offset=26),
                Call(
                  Name(
                    'enumerate',
                    Load(),
                    lineno=12,
                    col_offset=30,
                    end_lineno=12,
                    end_col_offset=39),
                  [
                    Name(
                      'matrix',
                      Load(),
                      lineno=12,
                      col_offset=40,
                      end_lineno=12,
                      end_col_offset=46)],
                  [],
                  lineno=12,
                  col_offset=30,
                  end_lineno=12,
                  end_col_offset=47),
                [],
                0),
              comprehension(
                Tuple(
                  [
                    Name(
                      'col_index',
                      Store(),
                      lineno=13,
                      col_offset=18,
                      end_lineno=13,
                      end_col_offset=27),
                    Name(
                      '_',
                      Store(),
                      lineno=13,
                      col_offset=29,
                      end_lineno=13,
                      end_col_offset=30)],
                  Store(),
                  lineno=13,
                  col_offset=18,
                  end_lineno=13,
                  end_col_offset=30),
                Call(
                  Name(
                    'enumerate',
                    Load(),
                    lineno=13,
                    col_offset=34,
                    end_lineno=13,
                    end_col_offset=43),
                  [
                    Subscript(
                      Name(
                        'matrix',
                        Load(),
                        lineno=13,
                        col_offset=44,
                        end_lineno=13,
                        end_col_offset=50),
                      Constant(
                        0,
                        lineno=13,
                        col_offset=51,
                        end_lineno=13,
                        end_col_offset=52),
                      Load(),
                      lineno=13,
                      col_offset=44,
                      end_lineno=13,
                      end_col_offset=53)],
                  [],
                  lineno=13,
                  col_offset=34,
                  end_lineno=13,
                  end_col_offset=54),
                [
                  Compare(
                    Subscript(
                      Name(
                        'mmax',
                        Load(),
                        lineno=14,
                        col_offset=17,
                        end_lineno=14,
                        end_col_offset=21),
                      Name(
                        'index',
                        Load(),
                        lineno=14,
                        col_offset=22,
                        end_lineno=14,
                        end_col_offset=27),
                      Load(),
                      lineno=14,
                      col_offset=17,
                      end_lineno=14,
                      end_col_offset=28),
                    [
                      Eq()],
                    [
                      Subscript(
                        Name(
                          'mmin',
                          Load(),
                          lineno=14,
                          col_offset=32,
                          end_lineno=14,
                          end_col_offset=36),
                        Name(
                          'col_index',
                          Load(),
                          lineno=14,
                          col_offset=37,
                          end_lineno=14,
                          end_col_offset=46),
                        Load(),
                        lineno=14,
                        col_offset=32,
                        end_lineno=14,
                        end_col_offset=47)],
                    lineno=14,
                    col_offset=17,
                    end_lineno=14,
                    end_col_offset=47)],
                0)],
            lineno=11,
            col_offset=13,
            end_lineno=14,
            end_col_offset=48),
          lineno=11,
          col_offset=4,
          end_lineno=14,
          end_col_offset=48),
        Return(
          BoolOp(
            Or(),
            [
              Name(
                'points',
                Load(),
                lineno=16,
                col_offset=11,
                end_lineno=16,
                end_col_offset=17),
              List(
                [],
                Load(),
                lineno=16,
                col_offset=21,
                end_lineno=16,
                end_col_offset=23)],
            lineno=16,
            col_offset=11,
            end_lineno=16,
            end_col_offset=23),
          lineno=16,
          col_offset=4,
          end_lineno=16,
          end_col_offset=23)],
      [],
      lineno=1,
      col_offset=0,
      end_lineno=16,
      end_col_offset=23)],
  [])
## END TREE BEFORE ##

## BEGIN NORMALIZED TREE ##
Module(
  [
    FunctionDef(
      'placeholder_0',
      arguments(
        [],
        [
          arg(
            'placeholder_1',
            lineno=1,
            col_offset=18,
            end_lineno=1,
            end_col_offset=24)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        If(
          UnaryOp(
            Not(),
            Name(
              'placeholder_1',
              Load(),
              lineno=2,
              col_offset=11,
              end_lineno=2,
              end_col_offset=17),
            lineno=2,
            col_offset=7,
            end_lineno=2,
            end_col_offset=17),
          [
            Return(
              List(
                [],
                Load(),
                lineno=3,
                col_offset=15,
                end_lineno=3,
                end_col_offset=17),
              lineno=3,
              col_offset=8,
              end_lineno=3,
              end_col_offset=17)],
          [],
          lineno=2,
          col_offset=4,
          end_lineno=3,
          end_col_offset=17),
        If(
          Call(
            Name(
              'any',
              Load(),
              lineno=5,
              col_offset=7,
              end_lineno=5,
              end_col_offset=10),
            [
              GeneratorExp(
                Compare(
                  Call(
                    Name(
                      'len',
                      Load(),
                      lineno=5,
                      col_offset=11,
                      end_lineno=5,
                      end_col_offset=14),
                    [
                      Name(
                        'placeholder_2',
                        Load(),
                        lineno=5,
                        col_offset=15,
                        end_lineno=5,
                        end_col_offset=18)],
                    [],
                    lineno=5,
                    col_offset=11,
                    end_lineno=5,
                    end_col_offset=19),
                  [
                    NotEq()],
                  [
                    Call(
                      Name(
                        'len',
                        Load(),
                        lineno=5,
                        col_offset=23,
                        end_lineno=5,
                        end_col_offset=26),
                      [
                        Subscript(
                          Name(
                            'placeholder_1',
                            Load(),
                            lineno=5,
                            col_offset=27,
                            end_lineno=5,
                            end_col_offset=33),
                          Constant(
                            0,
                            lineno=5,
                            col_offset=34,
                            end_lineno=5,
                            end_col_offset=35),
                          Load(),
                          lineno=5,
                          col_offset=27,
                          end_lineno=5,
                          end_col_offset=36)],
                      [],
                      lineno=5,
                      col_offset=23,
                      end_lineno=5,
                      end_col_offset=37)],
                  lineno=5,
                  col_offset=11,
                  end_lineno=5,
                  end_col_offset=37),
                [
                  comprehension(
                    Name(
                      'placeholder_2',
                      Store(),
                      lineno=5,
                      col_offset=42,
                      end_lineno=5,
                      end_col_offset=45),
                    Name(
                      'placeholder_1',
                      Load(),
                      lineno=5,
                      col_offset=49,
                      end_lineno=5,
                      end_col_offset=55),
                    [],
                    0)],
                lineno=5,
                col_offset=10,
                end_lineno=5,
                end_col_offset=56)],
            [],
            lineno=5,
            col_offset=7,
            end_lineno=5,
            end_col_offset=56),
          [
            Raise(
              Call(
                Name(
                  'ValueError',
                  Load(),
                  lineno=6,
                  col_offset=14,
                  end_lineno=6,
                  end_col_offset=24),
                [
                  Constant(
                    'irregular matrix',
                    lineno=6,
                    col_offset=25,
                    end_lineno=6,
                    end_col_offset=43)],
                [],
                lineno=6,
                col_offset=14,
                end_lineno=6,
                end_col_offset=44),
              lineno=6,
              col_offset=8,
              end_lineno=6,
              end_col_offset=44)],
          [],
          lineno=5,
          col_offset=4,
          end_lineno=6,
          end_col_offset=44),
        Assign(
          [
            Name(
              'placeholder_3',
              Store(),
              lineno=8,
              col_offset=4,
              end_lineno=8,
              end_col_offset=8)],
          ListComp(
            Call(
              Name(
                'max',
                Load(),
                lineno=8,
                col_offset=12,
                end_lineno=8,
                end_col_offset=15),
              [
                Name(
                  'placeholder_2',
                  Load(),
                  lineno=8,
                  col_offset=16,
                  end_lineno=8,
                  end_col_offset=19)],
              [],
              lineno=8,
              col_offset=12,
              end_lineno=8,
              end_col_offset=20),
            [
              comprehension(
                Name(
                  'placeholder_2',
                  Store(),
                  lineno=8,
                  col_offset=25,
                  end_lineno=8,
                  end_col_offset=28),
                Name(
                  'placeholder_1',
                  Load(),
                  lineno=8,
                  col_offset=32,
                  end_lineno=8,
                  end_col_offset=38),
                [],
                0)],
            lineno=8,
            col_offset=11,
            end_lineno=8,
            end_col_offset=39),
          lineno=8,
          col_offset=4,
          end_lineno=8,
          end_col_offset=39),
        Assign(
          [
            Name(
              'placeholder_4',
              Store(),
              lineno=9,
              col_offset=4,
              end_lineno=9,
              end_col_offset=8)],
          ListComp(
            Call(
              Name(
                'min',
                Load(),
                lineno=9,
                col_offset=12,
                end_lineno=9,
                end_col_offset=15),
              [
                Name(
                  'placeholder_5',
                  Load(),
                  lineno=9,
                  col_offset=16,
                  end_lineno=9,
                  end_col_offset=19)],
              [],
              lineno=9,
              col_offset=12,
              end_lineno=9,
              end_col_offset=20),
            [
              comprehension(
                Name(
                  'placeholder_5',
                  Store(),
                  lineno=9,
                  col_offset=25,
                  end_lineno=9,
                  end_col_offset=28),
                Call(
                  Name(
                    'zip',
                    Load(),
                    lineno=9,
                    col_offset=32,
                    end_lineno=9,
                    end_col_offset=35),
                  [
                    Starred(
                      Name(
                        'placeholder_1',
                        Load(),
                        lineno=9,
                        col_offset=37,
                        end_lineno=9,
                        end_col_offset=43),
                      Load(),
                      lineno=9,
                      col_offset=36,
                      end_lineno=9,
                      end_col_offset=43)],
                  [],
                  lineno=9,
                  col_offset=32,
                  end_lineno=9,
                  end_col_offset=44),
                [],
                0)],
            lineno=9,
            col_offset=11,
            end_lineno=9,
            end_col_offset=45),
          lineno=9,
          col_offset=4,
          end_lineno=9,
          end_col_offset=45),
        Assign(
          [
            Name(
              'placeholder_6',
              Store(),
              lineno=11,
              col_offset=4,
              end_lineno=11,
              end_col_offset=10)],
          ListComp(
            Dict(
              [
                Constant(
                  'row',
                  lineno=11,
                  col_offset=15,
                  end_lineno=11,
                  end_col_offset=20),
                Constant(
                  'column',
                  lineno=11,
                  col_offset=33,
                  end_lineno=11,
                  end_col_offset=41)],
              [
                BinOp(
                  Name(
                    'placeholder_7',
                    Load(),
                    lineno=11,
                    col_offset=22,
                    end_lineno=11,
                    end_col_offset=27),
                  Add(),
                  Constant(
                    1,
                    lineno=11,
                    col_offset=30,
                    end_lineno=11,
                    end_col_offset=31),
                  lineno=11,
                  col_offset=22,
                  end_lineno=11,
                  end_col_offset=31),
                BinOp(
                  Name(
                    'placeholder_9',
                    Load(),
                    lineno=11,
                    col_offset=43,
                    end_lineno=11,
                    end_col_offset=52),
                  Add(),
                  Constant(
                    1,
                    lineno=11,
                    col_offset=55,
                    end_lineno=11,
                    end_col_offset=56),
                  lineno=11,
                  col_offset=43,
                  end_lineno=11,
                  end_col_offset=56)],
              lineno=11,
              col_offset=14,
              end_lineno=11,
              end_col_offset=57),
            [
              comprehension(
                Tuple(
                  [
                    Name(
                      'placeholder_7',
                      Store(),
                      lineno=12,
                      col_offset=18,
                      end_lineno=12,
                      end_col_offset=23),
                    Name(
                      'placeholder_8',
                      Store(),
                      lineno=12,
                      col_offset=25,
                      end_lineno=12,
                      end_col_offset=26)],
                  Store(),
                  lineno=12,
                  col_offset=18,
                  end_lineno=12,
                  end_col_offset=26),
                Call(
                  Name(
                    'enumerate',
                    Load(),
                    lineno=12,
                    col_offset=30,
                    end_lineno=12,
                    end_col_offset=39),
                  [
                    Name(
                      'placeholder_1',
                      Load(),
                      lineno=12,
                      col_offset=40,
                      end_lineno=12,
                      end_col_offset=46)],
                  [],
                  lineno=12,
                  col_offset=30,
                  end_lineno=12,
                  end_col_offset=47),
                [],
                0),
              comprehension(
                Tuple(
                  [
                    Name(
                      'placeholder_9',
                      Store(),
                      lineno=13,
                      col_offset=18,
                      end_lineno=13,
                      end_col_offset=27),
                    Name(
                      'placeholder_8',
                      Store(),
                      lineno=13,
                      col_offset=29,
                      end_lineno=13,
                      end_col_offset=30)],
                  Store(),
                  lineno=13,
                  col_offset=18,
                  end_lineno=13,
                  end_col_offset=30),
                Call(
                  Name(
                    'enumerate',
                    Load(),
                    lineno=13,
                    col_offset=34,
                    end_lineno=13,
                    end_col_offset=43),
                  [
                    Subscript(
                      Name(
                        'placeholder_1',
                        Load(),
                        lineno=13,
                        col_offset=44,
                        end_lineno=13,
                        end_col_offset=50),
                      Constant(
                        0,
                        lineno=13,
                        col_offset=51,
                        end_lineno=13,
                        end_col_offset=52),
                      Load(),
                      lineno=13,
                      col_offset=44,
                      end_lineno=13,
                      end_col_offset=53)],
                  [],
                  lineno=13,
                  col_offset=34,
                  end_lineno=13,
                  end_col_offset=54),
                [
                  Compare(
                    Subscript(
                      Name(
                        'placeholder_3',
                        Load(),
                        lineno=14,
                        col_offset=17,
                        end_lineno=14,
                        end_col_offset=21),
                      Name(
                        'placeholder_7',
                        Load(),
                        lineno=14,
                        col_offset=22,
                        end_lineno=14,
                        end_col_offset=27),
                      Load(),
                      lineno=14,
                      col_offset=17,
                      end_lineno=14,
                      end_col_offset=28),
                    [
                      Eq()],
                    [
                      Subscript(
                        Name(
                          'placeholder_4',
                          Load(),
                          lineno=14,
                          col_offset=32,
                          end_lineno=14,
                          end_col_offset=36),
                        Name(
                          'placeholder_9',
                          Load(),
                          lineno=14,
                          col_offset=37,
                          end_lineno=14,
                          end_col_offset=46),
                        Load(),
                        lineno=14,
                        col_offset=32,
                        end_lineno=14,
                        end_col_offset=47)],
                    lineno=14,
                    col_offset=17,
                    end_lineno=14,
                    end_col_offset=47)],
                0)],
            lineno=11,
            col_offset=13,
            end_lineno=14,
            end_col_offset=48),
          lineno=11,
          col_offset=4,
          end_lineno=14,
          end_col_offset=48),
        Return(
          BoolOp(
            Or(),
            [
              Name(
                'placeholder_6',
                Load(),
                lineno=16,
                col_offset=11,
                end_lineno=16,
                end_col_offset=17),
              List(
                [],
                Load(),
                lineno=16,
                col_offset=21,
                end_lineno=16,
                end_col_offset=23)],
            lineno=16,
            col_offset=11,
            end_lineno=16,
            end_col_offset=23),
          lineno=16,
          col_offset=4,
          end_lineno=16,
          end_col_offset=23)],
      [],
      lineno=1,
      col_offset=0,
      end_lineno=16,
      end_col_offset=23)],
  [])
## END NORMALIZED TREE ##